src/Controls/src/Core/HandlerImpl/Window/Window.Impl.cs

```diff
@@ -561,7 +561,8 @@ private protected override void OnHandlerChangingCore(HandlerChangingEventArgs a
 
 			if (FlowDirection == FlowDirection.MatchParent && mauiContext != null)
 			{
-				FlowController.EffectiveFlowDirection = mauiContext.GetFlowDirection().ToEffectiveFlowDirection(true);
+				var flowDirection = Application?.PlatformLayoutDirection.ToFlowDirection() ?? FlowDirection.LeftToRight;
+				FlowController.EffectiveFlowDirection = flowDirection.ToEffectiveFlowDirection(true);
```

Comment:
Is this only called once per new `Window` created?

Reply:
See previous comment, it is also in the `void IApplication.ThemeChanged()` method.

Reply:
Reverting the layout direction changes as this is a larger thing and affect locale. Android supports configuration changes on Application and Activity, and each have different effects and requirements.

I still reverted the lazy layout direction in the AppInfo as this just fires once when the handler changes - which should basically be once. This code also means that the Activity will need to restart if the locale changes. And this is done automatically by the OS because we *don't* have the `ConfigChanges.Locale` on the `[Activity]`.